pkg/client/client.go

```diff
@@ -35,6 +35,11 @@ import (
 	"github.com/go-openapi/strfmt"
 )
 
+var (
+	// ciliumAgentTimeout is a timeout for cilium-agent readiness.
+	ciliumAgentTimeout = 30 * time.Second
```

Comment:
Call this `DefaultConnectTimeout`, make it a const and then use it from the CNI code as an argument to `NewDefaultClientWithTimeout()`

Reply:
Disregard my comments which you probably got on email - I forgot that with DefaultClient there is no possibility of getting errors due to parsing and every error should be related to connectivity.